mnist/main.py

```diff
@@ -100,10 +107,14 @@ def main():
     model = Net().to(device)
     optimizer = optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
 
-    for epoch in range(1, args.epochs + 1):
+    for epoch in range(1, args.epochs+1):
         train(args, model, device, train_loader, optimizer, epoch)
         test(args, model, device, test_loader)
 
+    if (args.save_model):
+        PATH="mnist_cnn.pt"
```

Comment:
additional PATH is unncessary, do `torch.save(model.state_dict(), "mnist_cnn.pt")`